hbase_sql/src/main/java/com/splicemachine/derby/impl/SpliceSpark.java

```diff
@@ -108,11 +111,30 @@ public static synchronized SparkSession getSessionUnsafe() {
             } catch (IOException e) {
                 throw new RuntimeException(e);
             }
-            System.exit(0);
-            session = initializeSparkSession();
+            if (isOlapWorker)
+                System.exit(0);
```

Comment:
Please add a LOG here so it's clear why we are shutting down

Reply:
Done.